master/internal/resourcemanagers/agent/slots.go

```diff
@@ -40,7 +40,13 @@ func (s *slots) Receive(ctx *actor.Context) error {
 			ctx.Tell(child, msg)
 		}
 	case aproto.ContainerStateChanged:
-		s.sendToSlots(ctx, msg.Container, msg)
+		if msg.Broadcast {
+			for _, child := range ctx.Children() {
```

Comment:
how does this come up? when would the container already be forgotten by the slot but we still attempt to recover it. i figure if the slot isn't aware of it, then it wouldn't be in the list of expected survivors?

Reply:
ah i see what you mean. i think the alternative makes sense.. i mean, a failure means we're already in a bad state i think. but i think this is fine. maybe just Broadcast -> BroadcastToAllSlots and leave a comment that in the event we couldn't recover the container we may not be sure which slots it was on (without more work) so we just tell all slots that a container they may have died.

Reply:
would still be nice to add this too, i think this would be a natural question to ask after reading the code (i could see a chesterton's fence situation where it gets removed by mistake)